### PR TITLE
Initialize build runner in AwtProcessor before usage

### PR DESCRIPTION
**Pull request by turing85**

### CODE REVIEW

**Comment by maxandersen** [comment]
can you give some context of why you need it quiet?

**Comment by turing85** [comment]
See [this zulip discussion](https://quarkusio.zulipchat.com/#narrow/stream/187038-dev/topic/What.20changed.20in.20GraalVM.2023.2E0.3F) for details.

The issue was already fixed in [this PR](https://github.com/quarkusio/quarkus/pull/34120).

File: extensions/awt/deployment/src/main/java/io/quarkus/awt/deployment/AwtProcessor.java [hunk 1]
```diff
@@ -95,7 +98,11 @@ ReflectiveClassBuildItem setupReflectionClassesWithMethods() {
     void setupAWTInit(BuildProducer<JniRuntimeAccessBuildItem> jc,
             BuildProducer<JniRuntimeAccessMethodBuildItem> jm,
             BuildProducer<JniRuntimeAccessFieldBuildItem> jf,
-            NativeImageRunnerBuildItem nativeImageRunnerBuildItem) {
+            NativeImageRunnerBuildItem nativeImageRunnerBuildItem,
+            Optional<ProcessInheritIODisabled> processInheritIODisabled,
+            Optional<ProcessInheritIODisabledBuildItem> processInheritIODisabledBuildItem) {
+        nativeImageRunnerBuildItem.getBuildRunner()
+                .setup(processInheritIODisabled.isPresent() || processInheritIODisabledBuildItem.isPresent());
```
Comment (gsmet) [inline extensions/awt/deployment/src/main/java/io/quarkus/awt/deployment/AwtProcessor.java:105]: If we have to do that, there is something wrong. I'll have to have a look later.

**Review by gsmet — CHANGES_REQUESTED** [review]

File: extensions/awt/deployment/src/main/java/io/quarkus/awt/deployment/AwtProcessor.java (hunk 1 above)
Comment (turing85) [reply extensions/awt/deployment/src/main/java/io/quarkus/awt/deployment/AwtProcessor.java:105]: Maybe I am misusing the `setup(...)` method, but I can say that `setup()` has not yet been called when `AwtProcessor` is executed. Without the call to `setup(...)`, the successive call to `nativeImageRunnerBuildItem.getBuildRunner().getGraalVMVersion()` would trigger the initial image pull.

**Comment by turing85** [comment]
@gsmet is this Pr still relevant, or can it be closed?

**Review by gsmet — APPROVED** [review]
I think what we are doing in `AwtProcessor` is not really in line with what we should do. Or we should initialize things differently.

Let's merge your PR if CI passes but ideally I think we should rework this to avoid having all these quirks in the build steps.

**Pull request merged**